### PR TITLE
add module to store ratings and cut computation time

### PR DESCRIPTION
**Pull request by JMaio**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -22,29 +22,38 @@
 #         [(0, 25), (1, 50), (2, 300)]]
 #
 
-#variables
+# variables
```
Comment (officialgupta) [inline main.py:25]: hahaha

File: main.py [hunk 2]
```diff
@@ -22,29 +22,38 @@
...
 cacheIndex = 0
 endpoints = []
 requests = []
+read_ratings_from_file = False
 
-#INPUT FROM FILE
-f = open("trending_today.in", "r", encoding="ascii")
+# INPUT FROM FILE
+filename = "trending_today"
```
Comment (officialgupta) [inline main.py:32]: good  idea

File: main.py [hunk 3]
```diff
@@ -83,39 +122,52 @@ def distro_weights(videos, requests):
...
 
 #FORMAT TO VALID
+#example caches array with 5 caches and the videos they store, some store just one video, some none and some more
+# caches = ((1,0),(2,3,4),(5),(),(5,4,6,3))
+
+if not read_ratings_from_file:
+    r = open(rating_file, "w")
+    rated_videos = rate_videos_bulk(videos, requests)
+    r.write(str(rated_videos))
+    r.close()
```
Comment (officialgupta) [inline main.py:162]: what is the rating_file for?

**Review by officialgupta — APPROVED** [review]

File: main.py (hunk 3 above)
Comment (JMaio) [reply main.py:162]: because rating videos always returns the same output, store that in a local file to test out the distribution algorithm immediately

**Pull request merged**